lib/gollum/views/helpers.rb

```diff
@@ -54,7 +54,9 @@ def self.included(base)
       end
 
       def page_route(page = nil)
-        clean_url(@base_url, page)
+        base_url = @base_url.nil? ? '/' : @base_url
+        base_url.prepend('/') unless base_url.start_with?('/')
+        clean_url(base_url, page)
```

Comment:
Does this really yield a string that starts with `'/'`? It seems the problem is that `clean_url` [removes all leading slashes](https://github.com/gollum/gollum/blob/master/lib/gollum/helpers.rb#L18). In that case, we could do:

```ruby
clean_url(@base_url ? @base_url : '/', page).prepend('/')
```

Or (faster):

```ruby
"/#{clean_url(@base_url ? @base_url : '/', page)}"
```

But perhaps the thing to do is first to write a test that defines the correct behavior?

Reply:
There are two `clean_url()` in gollum: one you mentioned and another just bellow `page_rouge()`.
https://github.com/gollum/gollum/blob/1d67565450679cb350937887eea608f48da57508/lib/gollum/views/helpers.rb#L56-L67
The latter one is used, and it just removes duplicated slashes. Maybe the function should be renamed. Or gollum should provide one and only way to handle `base_url` and `page_route` things, someday...

Reply:
Aah, thanks for the explanation!

Reply:
I would still like to avoid the `.prepend` call, as modifications to Strings are expensive in Ruby. What about simply:

```ruby
         clean_url("/#{@base_url}", page)
```

This would automatically handle the `nil` case:

```ruby
2.6.3 :001 > a = nil
 => nil 
2.6.3 :002 > "/#{a}"
 => "/" 
```

And since double slashes would be removed anyway by `clean_url`, I think we don't need to worry about the case in which `base_url.start_with?('/')`?

Reply:
Great! This is a good lesson for me.